identity/manager.go

```diff
@@ -45,23 +45,23 @@ func (idm *identityManager) GetIdentities() []dto.Identity {
 
 	var ids = make([]dto.Identity, len(accountList))
 	for i, account := range accountList {
-		ids[i] = *accountToIdentity(account)
+		ids[i] = accountToIdentity(account)
 	}
 
 	return ids
 }
 
-func (idm *identityManager) GetIdentity(identityString string) *dto.Identity {
+func (idm *identityManager) GetIdentity(identityString string) dto.Identity {
 	identityString = strings.ToLower(identityString)
 	for _, id := range idm.GetIdentities() {
 		if strings.ToLower(string(id)) == identityString {
-			return &id
+			return id
 		}
 	}
 
-	return nil
+	return dto.Identity("")
```

Comment:
`dto.Identity("")` is a valid identity, should probably return `nil, error`

Reply:
Yeah that's normal to return empty struct on error, if you return is by-value, not by-reference